debug ice: attempted to add with overflow

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code
`rustc --edition=2021`
```Rust
async fn main() {
    [0usize; 0xffff_ffff_ffff_ffff].await;
}

```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
c94cb834d05888ef5beeae2cc460156f435e2fb3
```

### Error output

```
error[E0277]: `[usize; usize::MAX]` is not a future
 --> D01319EFC4D41E1760BDE6D24D4EAEBDED06776A984903041C4E462ABADDEE9A.rs:2:37
  |
2 |     [0usize; 0xffff_ffff_ffff_ffff].await;
  |                                    -^^^^^
  |                                    ||
  |                                    |`[usize; usize::MAX]` is not a future
  |                                    help: remove the `.await`
  |
  = help: the trait `Future` is not implemented for `[usize; usize::MAX]`
  = note: [usize; usize::MAX] must be a future or must implement `IntoFuture` to be awaited
  = note: required for `[usize; usize::MAX]` to implement `IntoFuture`
```


<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"apekros"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs
+++ b/tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs
@@ -1,0 +1,10 @@
+// check-fail
+// edition:2021
+
+// test for issue-114912 - debug ice: attempted to add with overflow
+
+async fn main() {
+    //~^ ERROR `main` function is not allowed to be `async`
+    [0usize; 0xffff_ffff_ffff_ffff].await;
+    //~^ ERROR `[usize; usize::MAX]` is not a future
+}

--- a/tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.stderr
+++ b/tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.stderr
@@ -1,0 +1,23 @@
+error[E0277]: `[usize; usize::MAX]` is not a future
+  --> $DIR/debug-ice-attempted-to-add-with-overflow.rs:8:37
+   |
+LL |     [0usize; 0xffff_ffff_ffff_ffff].await;
+   |                                    -^^^^^
+   |                                    ||
+   |                                    |`[usize; usize::MAX]` is not a future
+   |                                    help: remove the `.await`
+   |
+   = help: the trait `Future` is not implemented for `[usize; usize::MAX]`
+   = note: [usize; usize::MAX] must be a future or must implement `IntoFuture` to be awaited
+   = note: required for `[usize; usize::MAX]` to implement `IntoFuture`
+
+error[E0752]: `main` function is not allowed to be `async`
+  --> $DIR/debug-ice-attempted-to-add-with-overflow.rs:6:1
+   |
+LL | async fn main() {
+   | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0277, E0752.
+For more information about an error, try `rustc --explain E0277`.

